src/System.Runtime/tests/System/Array.cs

```diff
@@ -307,6 +307,17 @@ public static void TestBinarySearchInRange<T>(T[] array, int index, int length,
     }
 
     [Fact]
+    public static void TestBinarySearch_Invalid()
+    {
+        Assert.Throws<ArgumentNullException>(() => Array.BinarySearch(null, 3));
+        Assert.Throws<ArgumentNullException>(() => Array.BinarySearch(null, 3, Comparer<int>.Default));
+        Assert.Throws<ArgumentNullException>(() => Array.BinarySearch(null, 0, 2, 3, Comparer<int>.Default));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.BinarySearch(new int[] { 1, 2 }, -1, 2, 3, Comparer<int>.Default));
+        Assert.Throws<ArgumentOutOfRangeException>(() => Array.BinarySearch(new int[] { 1, 2 }, 0, -1, 3, Comparer<int>.Default));
+        Assert.Throws<ArgumentException>(() => Array.BinarySearch(new int[] { 1, 2 }, 0, 3, 3, Comparer<int>.Default));
+    }
```

Comment:
Consider adding the argument name as the first argument to all of the Throws for argument exceptions.